backend.dev.Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM node:alpine3.17 AS builder
+WORKDIR /app
+
+COPY packages/server/package.json ./
+COPY yarn.lock ./
+RUN yarn remove node-cron @joystream/types
+RUN yarn --immutable
+
+COPY tsconfig.json ./base.tsconfig.json
+COPY packages/server/tsconfig.json ./
+RUN sed -i 's/"extends":.*/"extends": ".\/base.tsconfig.json",/' tsconfig.json
+
+COPY packages/server/prisma ./
+COPY packages/server/codegen.ts ./
+COPY packages/server/src ./src
+RUN yarn build:local
+
+RUN rm -rf node_modules
+ENV NODE_ENV=production
+RUN yarn --prod --immutable
+RUN yarn prisma generate
+
+FROM postgres:alpine3.17
```

Comment:
As I understand, this single image is supposed to operate both DB and the API for development. IMO this is a bit unintuitive and personally I'd prefer them separate and managed with `docker-compose`, like you would do for production setup.

Reply:
It makes sense but unless I forgot about something specific to the `:prod` image I think we could have a `docker-compose` file for development that uses this `:prod` image. But what I appreciate with the all bundled dev image is that it allows less experience developers to get started with a one liner (`docker run --rm -it -e POSTGRES_PASSWORD="my-password123" --name api -p 3000:3000 thesan/pioneer-backend`) regardless of the state of their local repository. This was really an issue back when I opened this PR (with only the development image).

But again I think it would make sense to add the `docker-compose` file add switch the `:latest` tag to the production image once we move it to `joystream/pioneer-backend`. WDYT ?

Reply:
I added the `docker-compose.yml` and made the `:prod` image the latest.